astropy/utils/decorators.py

```diff
@@ -493,8 +493,8 @@ class lazyproperty(property):
     the ``print`` statement is not executed.  Only the return value from the
     first access off ``complicated_property`` is returned.
 
-    If a setter for this property is defined, it will still be possible to
-    manually update the value of the property, if that capability is desired.
+    The lazyproperty also sets up a simple setter and deleter for the property
+    which can be overridden by own setter and deleter methods.
```

Comment:
I don't think this is much less confusing.  See my comment on the original issue.